lib/matplotlib/backends/_backend_tk.py

```diff
@@ -544,6 +544,22 @@ def __init__(self, canvas, window, *, pack_toolbar=True):
         if pack_toolbar:
             self.pack(side=tk.BOTTOM, fill=tk.X)
 
+    def _update_buttons_checked(self):
+        # sync button checkstates to match active mode
+        # Tk Checkbuttons only need to be switched off manually
+        # if mode was changed by some other button
```

Comment:
I think you might need to copy Qt here for selecting the buttons as well. The pan/zoom tools can be enabled by keyboard shortcut, and that is not reflected in the button state either.

Reply:
I honestly did not know about the keyboard shortcuts until now...